### PR TITLE
Django Frontend

### PR DESCRIPTION
**Pull request by chaten**

### CODE REVIEW

File: feats/django/templates/feats/detail.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        </dd>
+        {% endfor %}
+    </dl>
+</section>
+<section>
+    <h2>
+        Configuration
+    </h2>
+    {% if feature.feature.input_types %}
+	<section>
```
Comment (dbernard) [inline feats/django/templates/feats/detail.html:34]: Non-breaking but looks like theres some indentation mismatch going on here.

File: feats/django/templates/feats/segmentation/change_segments.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		var template = document.getElementById(templateId);
+		var container = document.getElementById(containerId);
+		var newRow = document.createElement("div")
+		var formNumber = Number(totalForms.value)
+		newRow.className = "form-group"
+		newRow.innerHTML = template.innerHTML.replace(/__prefix__/g, formNumber);
+		container.append(newRow);
+		totalForms.value = formNumber + 1;
+	}
+</script>
```
Comment (dbernard) [inline feats/django/templates/feats/segmentation/change_segments.html:15]: Looks like we use this same script block in a few places. Is there a way to pull this out into an include?

File: feats/django/views/segmentation.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+        return HttpResponseBadRequest()
+
+
+class ChangeSegmentation(base.TemplateView):
+    """
+    Changes how a Feature is Segmented and how selectors map to those segments.
+
+    Updating segmentation resets any selector mappings, so this form is done in
+    two stepsr
```
Comment (dbernard) [inline feats/django/views/segmentation.py]: Minor typo: `stepsr` --> `steps`

File: feats/selector.py [hunk 4]
```diff
@@ -88,20 +97,22 @@ def _hex_hash(self, key: str) -> str:
         ).hexdigest()
 
     def select(self, value: object) -> str:
-        key = self.segment(value)
+        key = self.segment.segment(value)
```
Comment (dbernard) [inline feats/selector.py:100]: So just to clarify, the first `segment` in `self.segment.segment(value)` is the capital "S" `Segment` object (noun), and the second `segment` is the verb segment (i.e. we're segmenting based on `value`), right?

Not trying to harp too hard on this - I know you've already mentioned possibly renaming segments anyways.

**Review by dbernard — COMMENTED** [review]
Couple low hanging comments to start on my first pass. Planning to go over this again shortly, but overall looking good.

File: feats/app.py [hunk 5]
```diff
@@ -43,6 +42,19 @@ def create(self, *args) -> object:
...
         return self.feature.implementations[impl_name].fn(*args)
 
+    def set_state(self, new_state: FeatureState):
+        serialized_state = copy.deepcopy(new_state.serialize(self.app))
+        self.app.storage[self.name].append(serialized_state)
+
+    def get_current_state(self) -> FeatureState:
+        states = self.app.storage[self.name]
+        try:
+            state_data = copy.deepcopy(states[-1])
```
Comment (dbernard) [inline feats/app.py:52]: Whats the driver behind using `deepcopy` in these two state methods?

File: feats/django/templates/feats/detail.html (hunk 1 above)
Comment (chaten) [reply feats/django/templates/feats/detail.html:34]: Yeah I haven't setup the same plugins on the home computer. I'll get this fixed

File: feats/django/views/segmentation.py (hunk 3 above)
Comment (chaten) [reply feats/django/views/segmentation.py]:
```suggestion
    two steps
```

File: feats/selector.py (hunk 4 above)
Comment (chaten) [reply feats/selector.py:100]: Yepppp 😬

File: feats/app.py (hunk 5 above)
Comment (chaten) [reply feats/app.py:52]: Without a deep copy, it is possible to modify the state without going through the set_state interface. This caused issues with the memory store & testing.

File: feats/django/templates/feats/segmentation/change_segments.html (hunk 2 above)
Comment (chaten) [reply feats/django/templates/feats/segmentation/change_segments.html:15]: I seem to have forgotten to delete some of the pages I refactored. I'm pretty sure this function is only used in one file.

File: feats/app.py (hunk 5 above)
Comment (dbernard) [reply feats/app.py:52]: That makes sense 👍

**Review by dbernard — APPROVED** [review]

**Pull request merged**